charms/istio-pilot/tests/unit/test_charm.py

```diff
@@ -477,3 +477,54 @@ def test_remove_exceptions(harness, mocked_client, mocker):
     mocked_client.return_value.delete.side_effect = api_error
     with pytest.raises(ApiError):
         harness.charm.on.remove.emit()
+
+
+#
+# Test Service
+#
+# Overriding mocked_get(autouse=True) fixture (disabling autouse) to allow setting
+# of return value of get() method.
+@pytest.fixture(scope="function")
+def mocked_get(mocked_charm_client, mocker):
+    return
```

Comment:
I don't understand why this is needed.  I'm also not sure how this affects the fixture.  Does this mean only usages of this fixture later in this file are affect?  Or is there some precedence order?

Reply:
This is to override `mocked_get(autouse=True)`.
I don't want `mocked_get()` to be patched for my particular test. However, it **is** patched with `autouse=True`. There is no way to disable `autouse` for just one test. As a result, this solution.

Reply:
An alternative to this could be to add a `yield mocked_charm_client` to the original `mocked_get` fixture, that way you can just edit its return here.  Although I don't know if that means you're editing the return for all uses, or just for this particular use...

Reply:
Actually, I see below you're editing the `.get.return_value` anyway - was that to avoid corrupting the global object?

Reply:
But by redefining it here, does it now make `mocked_get` not be an auto-used fixture for the other tests?  I worry that we might be disabling it for something else without realising

Reply:
> Actually, I see below you're editing the `.get.return_value` anyway - was that to avoid corrupting the global object?
Yes.
My concern is exactly what you described: I don't want to mess with other tests. That's why I've put `scope="function") for `mocked_get()` override.

Reply:
If this does override the autouse for all tests, then I think we'll accidentally break tests by doing this (otherwise, why would autouse be set for this fixture at all).  Thinking on this though, I mixed some things up - `autouse` and `scope` are different.  The autoused fixture is `scope=function` (that is the default), which means each test function call gets its own separate instance of the fixture.  So there should not be any bleed across tests.  You should be able to remove the override of mocked_get and leave your other code as is and it will have your desired effect.

Reply:
This might be a different problem if the scope was module or other, but the default is function so I don't think you need to work around things here

Reply:
Without override it does not allow me to set `harness.charm.lightkube_client.get.return_value`. Basically, without override, if I set `harness.charm.lightkube_client.get.return_value` to anything, then access property `_gateway` that uses `get()`, I always get `None`. Override solves this problem.

Reply:
Changed test to use `side_effect` and removed override of `mocked_get()`